lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/EvaluatorRequest.java

```diff
@@ -34,16 +37,30 @@
   private final int megaBytes;
   private final int number;
   private final int cores;
+  @Deprecated
   private final ResourceCatalog.Descriptor descriptor;
+  private final List<String> nodeNames;
+  private final List<String> rackNames;
 
+  @Deprecated
```

Comment:
Please add javadoc `@deprecated` as well and document when it was deprecated and what to use instead.